pupy/packages/all/pupyutils/search.py

```diff
@@ -8,11 +8,36 @@
 import mmap
 import threading
 import rpyc
+import pupy
+
+searchThread = None #If searchThread == None, no Search Thread is running
```

Comment:
This variable is module local. In case module updated thread will leak. If you want to have background searches then use Tasks please. If you find them not so perfect you may improve them.

Examples:
https://github.com/alxchk/pupy/blob/unstable/pupy/packages/linux/all/keylogger.py#L544
https://github.com/alxchk/pupy/blob/unstable/pupy/packages/linux/all/usniper.py#L14
https://github.com/alxchk/pupy/blob/unstable/pupy/packages/posix/all/hashmon.py#L38
https://github.com/alxchk/pupy/blob/unstable/pupy/packages/all/pupyutils/netcreds.py#L97

The idea is this object preserves result in some standard way, so you can acquire data with same API. Also this thing try to preserve results during shutdown (well, not always possible)